File: testbed_php/test/statemachine/FinalStateTest.php
<?php

class FinalStateTest extends UnitTestCase
{
    
  function test_CallDeleteOnceInFinalState()
  {
    $course = new CourseK();
    $course->flip();
    $this->assertEqual(1,$course->numberOfLogs());
    $this->assertEqual("deleted",$course->getLog(0));
  }
  
  function test_CallDeleteOnlyOnceAllFinalStatesAreReached()
  {
    $c = new CourseL();
    $this->assertEqual("StatusOn",$c->getStatus());
    $this->assertEqual("StatusOnMotorIdleMotorIdle",$c->getStatusOnMotorIdle());
    $this->assertEqual("StatusOnFanIdleFanIdle",$c->getStatusOnFanIdle());

    $c->flip();

    $this->assertEqual(1,$c->numberOfLogs());
    $this->assertEqual("deleted",$c->getLog(0));

    $this->assertEqual("StatusOn",$c->getStatus());
    $this->assertEqual("StatusOnMotorIdleFinal",$c->getStatusOnMotorIdle());
    $this->assertEqual("StatusOnFanIdleFanIdle",$c->getStatusOnFanIdle());
  }
  
}

File: testbed/test/cruise/statemachine/test/FinalStateTest.java
package cruise.statemachine.test;

import org.junit.*;

public class FinalStateTest
{

  @Test
  public void CallDeleteOnceInFinalState()
  {
    CourseK c = new CourseK();
    c.flip();
    Assert.assertEquals(1,c.numberOfLogs());
    Assert.assertEquals("deleted",c.getLog(0));
  }
  
  @Test
  public void CallDeleteWhenTransitionUsesFinalKeyword()
  {
    CourseL c = new CourseL();
    Assert.assertEquals(CourseL.Status.On,c.getStatus());
    Assert.assertEquals(CourseL.StatusOnMotorIdle.MotorIdle,c.getStatusOnMotorIdle());
    Assert.assertEquals(CourseL.StatusOnFanIdle.FanIdle,c.getStatusOnFanIdle());

    c.flip();

    Assert.assertEquals(1,c.numberOfLogs());
    Assert.assertEquals("deleted",c.getLog(0));

    Assert.assertEquals(CourseL.Status.Final,c.getStatus());
    Assert.assertEquals(CourseL.StatusOnMotorIdle.Null,c.getStatusOnMotorIdle());
    Assert.assertEquals(CourseL.StatusOnFanIdle.Null,c.getStatusOnFanIdle());
    
  }
  
  
}
